### PR TITLE
[INLONG-11028][SDK] Transform SQL support UrlEncode & UrlDecode functions

### LINKED ISSUE #11028: [Feature][SDK] Transform SQL support UrlEncode & UrlDecode functions
### Description

- parent issue https://github.com/apache/inlong/issues/10796
- URL_ENCODE(string):
  - Translates a string into ‘application/x-www-form-urlencoded’ format using the UTF-8 encoding scheme. If the input is NULL, or there is an issue with the encoding process, or the encoding scheme is not supported, will return NULL.
- URL_DECODE(string)
  - Decodes a given string in ‘application/x-www-form-urlencoded’ format using the UTF-8 encoding scheme. If the input is NULL, or there is an issue with the decoding process(such as encountering an illegal escape pattern), or the encoding scheme is not supported, the function returns NULL.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by MOONSakura0614**
Fixes #11028

### Motivation
Add 2 function class: UrlEncode & UrlDecode functions and their distinct unit tests.

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Review by luchunliang — APPROVED** [review]

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/UrlEncodeFunction.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            return null;
+        }
+    }
+
+    public static void main(String[] args) throws UnsupportedEncodingException {
+        String string = "https://www.google.com/search?q=java url encode";
+        String s = "bat apache";
+        System.out.println(URLEncoder.encode(string, StandardCharsets.UTF_8.toString()));
+        System.out.println(URLEncoder.encode(s, StandardCharsets.UTF_8.toString()));
+    }
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/UrlEncodeFunction.java:70]:
```suggestion
```

**Review by vernedeng — APPROVED** [review]

**Comment by MOONSakura0614** [comment]
@luchunliang @vernedeng @aloyszhang Sorry for demanding another review. I saw spotless process failed and recognized a useless import in a file so I redo the 'mvn spotless:apply' and re-commit.

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/UrlDecodeFunction.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        if (stringObj == null) {
+            return null;
+        }
+        String string = OperatorTools.parseString(stringObj);
+        if (string == null) {
+            return null;
+        }
+
+        try {
+            return URLDecoder.decode(string, StandardCharsets.UTF_8.toString());
```
Comment (luchunliang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/UrlDecodeFunction.java:57]: Miss the parameter of charset.
The default charset is UTF8.

**Review by aloyszhang — DISMISSED** [review]

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/UrlDecodeFunction.java (hunk 2 above)
Comment (MOONSakura0614) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/UrlDecodeFunction.java:57]: Thank you for your comment. I have specified the charset as `StandardCharsets.UTF_8.toString()` which effectively passes the UTF-8 charset. 

If your concern is about the use of `.toString()`, I can change it to `StandardCharsets.UTF_8.name()` for clarity. Let me know your preference.

**Review by luchunliang — APPROVED** [review]

**Comment by dockerzhang** [comment]
@MOONSakura0614 please rebase from the master branch to fix the failed workflow.

**Review by aloyszhang — DISMISSED** [review]

**Comment by aloyszhang** [comment]
UT failed, PTAL 
 TestUrlEncodeFunction.testUrlEncodeFunction:55 expected:<result=[]> but was:<result=[null]>

**Comment by MOONSakura0614** [comment]
fixed. Thank you for pointing the bug.

**Review by aloyszhang — APPROVED** [review]

**Review by luchunliang — APPROVED** [review]

**Pull request merged**